### ISSUE TITLE
ldapmod and ldap3 modules do not return memberOf filter results correctly

### ISSUE THREAD

**Issue by raddessi** [saltstack/salt#55370]
### Description of Issue
I have some users in groups, and I want to find out based on a list of groups names what the membership for each of those groups is. I can `ldapsearch` just fine, and I have verified both the ldap and ldap3 python modules (the ldap3 salt exec module doesn't even use ldap3?) return data correctly for me when run by hand. We are running freeipa and salt is able to communicate with the servers just fine, and can do a very simple `cn=` filter match as well.

### Setup
Standard ldap setup, not much I can provide here that wouldn't be redacted.

### Steps to Reproduce Issue
`salt 'minion' ldap3.search "{'url': 'ldap://<our server>/'}" "base=<our base dn>" scope="subtree" filterstr=memberOf=<our group dn>` returns no data.

`ldapsearch -x -D "<bind user>" -w '<bind password>' -h <our server> -b "<our users dn>" '(memberof=<our group dn>)'`  returns the data we expect to see.

The manual python ldap module test also returns correctly:
```python
l = ldap.initialize("ldap://<our server>/")
l.simple_bind('<bind user>', '<bind password>')
l.search_s("<our users dn>", scope=2, filterstr="memberOf=<our group dn>")
```

I added logging to ldap3.py and the call to l.c.search_s is being populated with the correct data there as well so there must be something in the binding or setopt setup that is causing this? I'm at a loss.

### Versions Report
```
Salt Version:
           Salt: 2019.2.0
 
Dependency Versions:
           cffi: 1.12.2
       cherrypy: unknown
       dateutil: 2.8.0
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.11
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.26.6
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.4
         Python: 3.6.6 | packaged by conda-forge | (default, Oct 12 2018, 14:43:46)
   python-gnupg: 0.4.4
         PyYAML: 5.1
          PyZMQ: 18.0.1
           RAET: Not Installed
          smmap: 2.0.5
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1
 
System Versions:
           dist: centos 7.6.1810 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-957.12.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.6.1810 Core

```

**Comment by raddessi** [comment]
This got marked as stale before anyone worked on it afaik. Bumping, it's still an issue

**Comment by raddessi** [comment]
Is bumping acceptable here?

**Comment by sagetherage** [comment]
@raddessi is still an issue on supported versions of Salt?

**Comment by raddessi** [comment]
I'll check by some time next week and get back to you

**Comment by raddessi** [comment]
Confirmed it's still an issue as of 3003. 

# test steps
## ldapsearch
`ldapsearch -x -D "<bind user>" -w '<bind password>' -h <our server> -b "<our users dn>" '(memberof=<our group dn>)'` still returns the data we expect to see from using the filter `(memberof=<our group dn>)`

## salt
1. `conda create -n salt55370 python=3.8 -y`
2. `conda activate salt55370`
3. `pip install salt==3003 python-ldap==3.3.1`
4. `salt 'minion' ldap3.search "{'url': 'ldap://<our server>/'}" "base=<our base dn>" scope="subtree"` returns all of the users like we would expect since there is no filter applied
5. `salt 'minion' ldap3.search "{'url': 'ldap://<our server>/'}" "base=<our base dn>" scope="subtree" filterstr=memberOf=<our group dn>` returns no data
6. `pip install ipython`
7. `ipython`
    ```python
    l = ldap.initialize("ldap://<our server>/")
    l.simple_bind('<bind user>', '<bind password>')
    l.search_s("<our users dn>", scope=2, filterstr="memberOf=<our group dn>")
    ```
    This will print out the filtered list of users as expected.

If I can provide anything else let me know.

**Comment by rhansen** [comment]
@raddessi I noticed that your steps to reproduce do `filterstr="memberOf=<our group dn>"` rather than `filterstr="(memberOf=<our group dn>)"`.  If you add the parentheses does it work?

**Comment by raddessi** [comment]
I checked my notes and yes I was using parentheses. I tested again on `v3004` and it is still broken with and without parentheses.

**Comment by raddessi** [comment]
Oh wow, I figured this out. This may have been on my end. I have in my notes that this code should have been picking up the minion's ldap config DN and password from the minion config file but it seems it may not be. When passing in ` "{'url': 'ldap://<server>/', 'bind': {'method': 'simple', 'dn': '<redacted>', 'password': '<redacted>'}}"` instead of `"{'url': 'ldap://<our server>/'}"` it now functions as expected. I'm fairly sure I would have tested this when I opened this ticket but it's too far back for me to remember now. 

Either way I think that this can be marked as resolved. Even the normal `ldapmod` module functions for me now (I do have that it did not before) with config put in the minion file. Thanks for the ping!
